### DISCUSSION TITLE
Database choice

### DISCUSSION THREAD

**Discussion by Ojani** [uprm-inso4101-2024-2025-s1/semester-project-campus-resource-and-event-management#37] [Reminders]
We haven't spoken about databases, and while I did suggest making a dummy database interphase in another discussion post so it could be easy to plug in a database later on, we eventually need to decide on an actual database, wether we manage it ourselves or opt in for a cloud solution. There are many options, I suggest using this discussion to talk about them.

**Comment by SaucyRexy** [comment]
Here are some free database possibilities that we can use for the start of the project:

1. PostgreSQL

Why: PostgreSQL is a powerful, open-source relational database that’s ideal for handling structured data like user profiles, event details, and bookings. It's free, highly reliable, and supports complex queries, which will be helpful for analyzing resource usage and event participation.

2. MySQL/MariaDB
 
Why: Both MySQL and its fork MariaDB are popular relational databases, also open-source and free to use. They are known for being fast, easy to set up, and well-suited for web applications that need consistent performance.

3. Firebase Firestore

Why: Firebase Firestore is a NoSQL cloud database that’s great for apps requiring real-time data updates. It’s free to start with, though it can get expensive at scale. It’s a good option if we want features like real-time resource availability and instant event notifications.

4. MongDB
 
Why: MongoDB is another open-source NoSQL database, ideal for storing flexible, unstructured data. It’s free under the Community Edition and would be useful if our data structures need to evolve over time.

5. SQLite

Why: SQLite is a super lightweight, serverless database that's free and easy to set up. It might be a good option for initial development or prototyping, though it's not ideal for large-scale production use.

Given that we’re still in the development phase and working with limited resources, starting with PostgreSQL, MySQL/MariaDB, or SQLite could be our best bet. They’re all free and widely supported, and they offer the reliability we need for handling structured data. As we scale up, we could consider cloud solutions like Firebase Firestore or MongoDB, depending on our needs.

Let me know what you all think or if there are other options we should consider!
